### DISCUSSION TITLE
Support Default value for unselected columns

### DISCUSSION THREAD

**Discussion by Reverier-Xu** [SeaQL/sea-orm#1507] [Ideas]
## Motivation

I have read your document [CustomSelect](https://www.sea-ql.org/SeaORM/docs/advanced-query/custom-select/) and some issues (#302), it seems that if I want to select only some of the fields, I have to define a corresponding Lite Struct to receive the data returned by Sea Orm. This can be troublesome in some cases, for example, in a User table with many information fields, I only want to hide the password field, in order to filter out this field I need to define two very large Model structs, the difference between them is only that one of them does not have the password field.

## Proposed Solutions

Is it possible to have the unselected fields automatically populated with `Default::default()` or some other default value implementation?

## Current Workarounds

none.

**Comment by Reverier-Xu** [comment]
Hi, maybe we can provide a macro like this:

```rust
#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
#[sea_orm(table_name = "cake")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    #[sea_orm(allow_missing)] // <- allow the default value to be filled when a `ColumnNotFound` error occurs in `sqlx::*Row::try_get`
    pub taste_description: String,
}
```

and then modify the error handling in `src/executor/query.rs` `try_getable_*` to achieve this?

This solution doesn't look clean, but if it's okay, I'll work on it.

**Reply by tyt2y3** [reply]
I think you just have to make it an Option String and unwrap_or_default on it

**Reply by Reverier-Xu** [reply]
Okay, see #1513
